### PR TITLE
add cronjob resource name for get -h

### PR DESCRIPTION
**Pull request by shiywang**
Fixes https://github.com/kubernetes/kubernetes/issues/43293
@gyliu513 @ymqytw ptal

### CODE REVIEW

**Comment by shiywang** [comment]
I'm using `kubectl get /tab` completion  command there is 37 resources which can get, but currently there only 34 resources,  I also sort the help info follow the alpha order

File: pkg/kubectl/cmd/cmd.go [hunk 1]
```diff
@@ -183,11 +183,12 @@ __custom_func() {
 
     * all
     * certificatesigningrequests (aka 'csr')
-    * clusters (valid only for federation apiservers)
     * clusterrolebindings
     * clusterroles
+    * clusters (valid only for federation apiservers)
     * componentstatuses (aka 'cs')
     * configmaps (aka 'cm')
+    * cronjob
```
Comment (gyliu513) [inline pkg/kubectl/cmd/cmd.go]: I prefer that we use `cronjobs` here to keep consistent.

**Comment by ymqytw** [comment]
@k8s-bot ok to test

**Comment by ymqytw** [comment]
Please file another PR to make sure the resources in https://github.com/kubernetes/kubernetes.github.io/blob/master/docs/user-guide/kubectl-overview.md are synced.

File: pkg/kubectl/cmd/cmd.go [hunk 2]
```diff
@@ -214,7 +215,9 @@ __custom_func() {
     * serviceaccounts (aka 'sa')
     * services (aka 'svc')
     * statefulsets
+    * statuses
```
Comment (ymqytw) [inline pkg/kubectl/cmd/cmd.go]: I don't think we need this.
kubectl cannot get this 
```
$ kubectl get status 
the server doesn't have a resource type "status"
```

File: pkg/kubectl/cmd/cmd.go [hunk 3]
```diff
@@ -214,7 +215,9 @@ __custom_func() {
...
     * storageclasses
+    * thirdpartyresourcedata
```
Comment (ymqytw) [inline pkg/kubectl/cmd/cmd.go]: Same for this one.

**Comment by ymqytw** [comment]
/release-note-none
@adohe @pwittrock Can we get this to 1.6? This is just a help message update.

**Comment by shiywang** [comment]
@ymqytw @adohe @gyliu513 ptal, thanks

**Comment by shiywang** [comment]
@ymqytw @adohe @gyliu513 ptal, thanks

File: pkg/kubectl/cmd/cmd.go [hunk 4]
```diff
@@ -183,11 +183,12 @@ __custom_func() {
 
     * all
     * certificatesigningrequests (aka 'csr')
-    * clusters (valid only for federation apiservers)
     * clusterrolebindings
     * clusterroles
+    * clusters (valid only for federation apiservers)
     * componentstatuses (aka 'cs')
     * configmaps (aka 'cm')
+    * cronjobs
```
Comment (gyliu513) [inline pkg/kubectl/cmd/cmd.go]: I'd like we also put a short name for this, such as `cjobs` etc. But we can fix this in a follow up patch.

**Comment by gyliu513** [comment]
lgtm, thanks @shiywang

**Comment by shiywang** [comment]
@k8s-bot gce etcd3 e2e test this

File: pkg/kubectl/cmd/cmd.go (hunk 3 above)
Comment (adohe) [inline pkg/kubectl/cmd/cmd.go]: nice catch.

File: pkg/kubectl/cmd/cmd.go (hunk 4 above)
Comment (adohe) [inline pkg/kubectl/cmd/cmd.go]: pls don't do this here.

**Comment by shiywang** [comment]
@k8s-bot cvm gce e2e test this

**Comment by shiywang** [comment]
@k8s-bot non-cri e2e test this

**Pull request merged**